src/vegasflow/vflowplus.py

```diff
@@ -218,6 +200,11 @@ def _iteration_content(self):
             self.refine_grid(arr_res2)
         return res, sigma
 
+    def run_event(self, tensorize_events=True, **kwargs):
+        """ Tensorizes the number of events so they are not python or numpy primitives """
+        logger.warning("Variable number of events requires function signatures all across")
+        return super().run_event(tensorize_events=tensorize_events, **kwargs)
+
```

Comment:
If I understand correctly now every integration with `VegasFlowPlus` will use this `run_event` with `tensorize_events=True` to handle different number of calls.
By setting `self.adaptive=False`, therefore using standard vegas,  the number of calls is fixed in every iteration and there are not retracing problems.
Maybe we can use `self.adaptive` in order to choose to tensorize `ncalls` if `self.adaptive = True` or to keep it as a python/numpy primitive if `self.adaptive = False`?

Reply:
Yes. You are correct. The right thing would be to write something like:

```python
    def __init__(  ... ):
        # <code from the init here>
        if adaptive:
              logger.warning("When adaptive is active all compiled functions require signature")

    def run_event(self, tensorize_events=None, **kwargs):
        """ Tensorizes the number of events so they are not python or numpy primitives if adaptive=True"""
        return super().run_event(tensorize_events=self.adaptive, **kwargs)
```

Reply:
Ok, perfect!
I will change it now.
Thanks!